File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/users/manager/MyAvatarState.java
package com.lumiyaviewer.lumiya.slproto.users.manager;

public abstract class MyAvatarState {
    public static MyAvatarState create(boolean z, int i, boolean z2, boolean z3) {
        return new AutoValue_MyAvatarState(z, i, z2, z3);
    }

    public abstract boolean hasHUDs();

    public abstract boolean isFlying();

    public abstract boolean isSitting();

    public abstract int sittingOn();
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/users/manager/MyAvatarState.kt
package com.lumiyaviewer.lumiya.slproto.users.manager

abstract class MyAvatarState {
    abstract fun hasHUDs(): Boolean
    abstract fun isFlying(): Boolean
    abstract fun isSitting(): Boolean
    abstract fun sittingOn(): Int

    companion object {
        @JvmStatic fun create(flying: Boolean, sittingOn: Int, sitting: Boolean, hasHUDs: Boolean): MyAvatarState =
            AutoValue_MyAvatarState(flying, sittingOn, sitting, hasHUDs)
    }
}
